Dockerfile

```diff
@@ -1,12 +1,23 @@
-FROM node:lts-slim  AS runner
+FROM node:lts-slim AS runner
+
+ENV OPAMYES=1 \
+    NO_UPDATE_NOTIFIER=true
 
 RUN apt-get update && \
-    apt-get install -y jq && \
+    apt-get install -y --no-install-recommends jq curl git opam ca-certificates && \
     apt-get purge --auto-remove -y && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/*
 
-ENV NO_UPDATE_NOTIFIER=true
+# Initialize OPAM
+RUN opam init --bare --disable-sandboxing --shell-setup && \
+    opam switch create default 5.3.0 && \
+    eval $(opam env --switch=default)
+
+RUN opam update
+RUN opam install dune reason melange melange-jest
+
+RUN apt-get remove -y curl git ca-certificates
```

Comment:
Should the `apt-get purge --auto-remove && apt-get clean` happen after the `apt-get remove`?

Reply:
Thanks, will give it a try